docs/source/dev_guide/pre_release_testing.rst

```diff
@@ -0,0 +1,59 @@
+     git checkout -b test_pre_rc<#> upstream/main
+     git log # check the commits match https://github.com/E3SM-Project/zppy/commits/main
+
+4. Set up a dev environment for E3SM Diags. ::
+
+     # `cd` to e3sm_diags directory
+     $ git checkout main
+     $ git fetch upstream
+     $ git reset --hard upstream/main
+     $ git log # Should match https://github.com/E3SM-Project/e3sm_diags/commits/main
+     $ mamba clean --all
+     $ mamba env create -f conda-env/dev.yml -n e3sm_diags_<date>
+     $ conda activate e3sm_diags_<date>
+     $ pip install .
+
+5. Make sure you're using the latest packages. In ``tests/integration/utils.py``, do the following:
+
+   * Change the last line ``generate_cfgs()`` to ``generate_cfgs(unified_testing=True)``. If this is not done, then ``zppy`` will simply load the latest officially released E3SM Unified environment by default.
+   * Update ``environment_commands_test`` for the machine you're currently working on. If this is not done, then ``zppy`` will use the existing value, which is likely an old E3SM Unified RC launch script that doesn't even exist anymore. This parameter should be set to the latest E3SM Unified RC launch script, since E3SM Unified is the only environment that all ``zppy`` tasks can run from. For instance, ``ncclimo`` isn't in the ``zppy`` dev environment.
+   * Update ``diags_environment_commands`` to use the environment created in the above step. If this is not done, the ``atm_monthly_180x360_aave_environment_commands`` subtask will use an older dev environment for E3SM Diags than we want.
+     
+6. Set up a new development environment. This ensures that testing will use the latest conda changes. Note that you will need to run ``conda remove -n zppy_dev_pre_rc<#> --all`` first if you have previously done this step. ::
+
+     mamba clean --all
+     mamba env create -f conda/dev.yml -n zppy_dev_pre_rc<#>
+     conda activate zppy_dev_pre_rc<#>
+     pip install .
+
+7. Run the unit tests with ``python -u -m unittest tests/test_*.py``.
+
+   * If there are any failures, fix the code (or tests), and then go back to the above step.
+   * If there are no failures, proceed to the next step.
+     
+8. Run the "Commands to run before running integration tests" for the current machine. To ensure you don't encounter issues from running ``zppy`` commands simultaneously, wait for all the automatically launched jobs from one run to finish before running ``zppy`` again.
+
+   * `Chrysalis <https://github.com/E3SM-Project/zppy/blob/main/tests/integration/generated/directions_chrysalis.md>`_
+   * `Compy <https://github.com/E3SM-Project/zppy/blob/main/tests/integration/generated/directions_compy.md>`_
+   * `Perlmutter <https://github.com/E3SM-Project/zppy/blob/main/tests/integration/generated/directions_pm-cpu.md>`_
+
+9. Run the integration tests with ``python -u -m unittest tests/integration/test_*.py``. Note that ``test_complete_run.py`` takes approximately 75 minutes to run on Compy.
```

Comment:
About 20 minutes on Chrysalis. The `zppy` run of `test_complete_run_chrysalis.cfg` takes a little over an hour.

Reply:
Compy: `Ran 13 tests in 5836.885s` [97.3 minutes] for `python -u -m unittest tests/integration/test_*.py`

Reply:
Compy, after updating expected files: `Ran 1 test in 669.627s` [11.2 minutes] for `python -u -m unittest tests/integration/test_complete_run.py` (the longest-running test).

I suspect the printing of error messages may be what slowed the previous run down so much.

Reply:
Compy, after updating expected files: `Ran 13 tests in 2854.441s` [47.6 minutes] for `python -u -m unittest tests/integration/test_*.py` (all integration tests)